### DISCUSSION TITLE
Buildpacks: Best practices and Guidelines

### DISCUSSION THREAD

**Discussion by sambhav** [buildpacks/community#89] [Ideas]
Hello all, as a part of [RFC #83](https://github.com/buildpacks/rfcs/blob/main/text/0083-best-practices-and-guidelines.md) we will be creating a best practices and guidelines section in the [buildpacks.io](https://buildpacks.io/) website. Initially, we are trying to seed these guidelines from experienced buildpack authors, operators and platform maintainers. If you have any suggestions for the guidelines, please add them to the discussion below in the following format - 
```markdown
Scope: Select one of <App developer, Buildpack Author, Operator, Platform maintainer> 
Title: A small descriptive title 
Guideline: A long form description of the guideline
Rationale: The rationale behind the guideline
Examples: Some examples of the guideline in practice
```

**Comment by hone** [comment]
This coming from the leadership meeting today and maybe not what you were looking for:

Scope: Buildpack Author
Title: Build Plan Usage Guidelines
Guideline: TBD
Rationale: When explaining buildpacks, the Build Plan is often confusing. I personally always have to look up in the spec if `metadata` should be used in `require` or `provide`. A guide on how it can be used to coordinate between multiple buildpacks would go a long way. In addition, figuring out a standard on how names should be handled/used since they're "just strings" at this point. There is potential for "accidental conflicts".
Examples: A description of how paketo and other folks are using the Build Plan.
